### PR TITLE
Fix CUDA 11 multi-GPU FFT bug

### LINKED ISSUE #3757: CUDA 11 Test: `cupy_tests.fft_tests.test_fft.TestMultiGpu`
```
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_2_{n=None, norm=None, shape=(64,)}::test_fft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_2_{n=None, norm=None, shape=(64,)}::test_ifft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_6_{n=None, norm='ortho', shape=(64,)}::test_fft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_6_{n=None, norm='ortho', shape=(64,)}::test_ifft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_24_{n=64, norm=None, shape=(0,)}::test_fft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_24_{n=64, norm=None, shape=(0,)}::test_ifft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_26_{n=64, norm=None, shape=(64,)}::test_fft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_26_{n=64, norm=None, shape=(64,)}::test_ifft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_28_{n=64, norm='ortho', shape=(0,)}::test_fft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_28_{n=64, norm='ortho', shape=(0,)}::test_ifft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_30_{n=64, norm='ortho', shape=(64,)}::test_fft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuFft_param_30_{n=64, norm='ortho', shape=(64,)}::test_ifft - AssertionError: Parameterized test failed.
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_0_{n=None, norm=None, shape=(64,)}::test_fft - AssertionError: Parameterized test f...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_0_{n=None, norm=None, shape=(64,)}::test_ifft - AssertionError: Parameterized test ...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_1_{n=None, norm=None, shape=(128,)}::test_fft - AssertionError: Parameterized test ...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_1_{n=None, norm=None, shape=(128,)}::test_ifft - AssertionError: Parameterized test...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_2_{n=None, norm='ortho', shape=(64,)}::test_fft - AssertionError: Parameterized tes...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_2_{n=None, norm='ortho', shape=(64,)}::test_ifft - AssertionError: Parameterized te...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_3_{n=None, norm='ortho', shape=(128,)}::test_fft - AssertionError: Parameterized te...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_3_{n=None, norm='ortho', shape=(128,)}::test_ifft - AssertionError: Parameterized t...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_4_{n=64, norm=None, shape=(64,)}::test_fft - AssertionError: Parameterized test fai...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_4_{n=64, norm=None, shape=(64,)}::test_ifft - AssertionError: Parameterized test fa...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_5_{n=64, norm=None, shape=(128,)}::test_fft - AssertionError: Parameterized test fa...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_5_{n=64, norm=None, shape=(128,)}::test_ifft - AssertionError: Parameterized test f...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_6_{n=64, norm='ortho', shape=(64,)}::test_fft - AssertionError: Parameterized test ...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_6_{n=64, norm='ortho', shape=(64,)}::test_ifft - AssertionError: Parameterized test...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_7_{n=64, norm='ortho', shape=(128,)}::test_fft - AssertionError: Parameterized test...
FAILED tests/cupy_tests/fft_tests/test_fft.py::TestMultiGpuPlanCtxManagerFft_param_7_{n=64, norm='ortho', shape=(128,)}::test_ifft - AssertionError: Parameterized tes...
```

### PR DESCRIPTION
**Pull request by leofang**
Close #3757. See https://github.com/cupy/cupy/issues/3757#issuecomment-673184478 for the trigger condition. 

cc: @kmaehashi @pentschev @anaruse @maxpkatz

### CODE REVIEW

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by pentschev** [comment]
For visibility, as per https://github.com/cupy/cupy/issues/3757#issuecomment-673356071 I can reproduce this error with the CUDA 11.0 `simpleCUFFT_MGPU` sample and I'm raising it to cuFFT developers.

**Comment by pentschev** [comment]
I filed an internal bug with ID `3093094`, in case you want to add a comment there @leofang .

**Comment by leofang** [comment]
Thanks, @pentschev. I added the ticket number to the comment.

Jenkins, test this please

File: tests/cupy_tests/fft_tests/test_fft.py [hunk 1]
```diff
@@ -159,6 +161,17 @@ def test_ifft(self, xp, dtype):
         return out
 
 
+# See #3757 and NVIDIA internal ticket 3093094
+def _skip_multi_gpu_bug(shape, gpus):
+    # avoid CUDA 11 bug triggered by
```
Comment (pentschev) [inline tests/cupy_tests/fft_tests/test_fft.py:166]:
```suggestion
    # avoid CUDA 11.0 (will be fixed by CUDA 11.2) bug triggered by
```

File: tests/cupy_tests/fft_tests/test_fft.py [hunk 2]
```diff
@@ -159,6 +161,17 @@ def test_ifft(self, xp, dtype):
...
+    # - batch = 1
+    # - gpus = [1, 0]
+    if (cupy.cuda.runtime.runtimeGetVersion() == 11000
```
Comment (pentschev) [inline tests/cupy_tests/fft_tests/test_fft.py:169]:
```suggestion
    if (cupy.cuda.runtime.runtimeGetVersion() == 11200
```

Comment (pentschev) [reply tests/cupy_tests/fft_tests/test_fft.py:169]: Originally I forgot to change the comparison operator, which I fixed now.

**Comment by leofang** [comment]
Sounds good. Thanks @pentschev! Let's do this. If they don't fix this by 11.2 we will know! 😂

File: tests/cupy_tests/fft_tests/test_fft.py (hunk 2 above)
Comment (leofang) [reply tests/cupy_tests/fft_tests/test_fft.py:169]: I think you meant this?
```suggestion
    if (11000 <= cupy.cuda.runtime.runtimeGetVersion() < 11200
```

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Restarted CI as something went wrong with docker build:
https://jenkins.preferred.jp/job/chainer/job/cupy_pr/1769/TEST=cupy-py35,label=mn1-p100/console

**Comment by leofang** [comment]
@kmaehashi cupy-py35 is naughty today 😂

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
It seems network connection is unstable or extremely slow so CUDA download is failing... I'll take a look on it

**Comment by pentschev** [comment]
> Sounds good. Thanks @pentschev! Let's do this. If they don't fix this by 11.2 we will know! 😂

Precisely my point! 😄

File: tests/cupy_tests/fft_tests/test_fft.py (hunk 2 above)
Comment (pentschev) [reply tests/cupy_tests/fft_tests/test_fft.py]: Ah yes, totally missed that, thanks for fixing!

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Thank you, LGTM!

**Comment by kmaehashi** [comment]
backport not needed as multi-gpu fft feature is not backported to v7

**Comment by leofang** [comment]
Thanks @pentschev @kmaehashi!

**Comment by pentschev** [comment]
I was provided with a preview build of cuFFT 11.2 and was able to verify that the issue is indeed fixed in that version, the condition in https://github.com/cupy/cupy/pull/3775/files#diff-ccb51faf00a37e7d1236eefcdef3d78aaba921baeca21e7405169a4c43d9398bR169 should then be fine and no other action should be needed on CuPy's end. Thanks everyone!

**Pull request merged**